R/frequencies_to_stop_times.R

```diff
@@ -0,0 +1,112 @@
+#' frequencies_to_stop_times
+#' 
+#' Convert gtfs frequencies to stop_times
+#'
+#' @param gtfs A set of GTFS data returned from \link{extract_gtfs}.
+#'
+#' @return The input gtfs data with data from `frequencies` replaced by
+#' arrival_time and departure_time in `stop_times`
+#' 
+#' @importFrom data.table shift
+#'    
+#' @export
+frequencies_to_stop_times <- function(gtfs)  
+{
+    # check if gtfs is a gtfs class of object
+    if (class(gtfs)[1] != "gtfs")
+        stop ("selected object does not appear to be a GTFS file")
+    
+    # test if gtfs contains no empty `frequencies`
+    if (!("frequencies" %in% names (gtfs)))
+        stop ("selected gtfs does not contain frequencies")
+    if (nrow (gtfs$frequencies) == 0)
+        stop ("frequencies table is empty")
+    
+    # frequencies must contain required columns:
+    # "trip_id" ,"start_time","end_time","headway_secs"
+    need_these_columns <- c("trip_id" ,"start_time","end_time","headway_secs")
+    checks <- vapply (need_these_columns, function (i)
+        any (grepl (i, names(gtfs$frequencies))), logical (1))
+
```

Comment:
@stmarcin Can you please add the following lines somewhere like around this spot, in order to suppress the NOTEs about "no visible binding":
``` r
arrival_time <- departure_time <- end_time <-
  start_time <- stop_sequence <- trip_id <- trip_id_f <- NULL
```